shotover-proxy/src/transforms/tee.rs

```diff
@@ -128,11 +128,11 @@ impl Transform for Tee {
                 let chain_response = chain_result?;
 
                 if !chain_response.eq(&tee_response) {
-                    Ok(vec!(Message::new(
-                        MessageDetails::Response(QueryResponse::empty_with_error(Some(Value::Strings(
-                            "The responses from the Tee subchain and down-chain did not match and behavior is set to fail on mismatch"
+                    Ok(vec!(Message::new_response(
+                        QueryResponse::empty_with_error(Some(Value::Strings(
+                            "ERR The responses from the Tee subchain and down-chain did not match and behavior is set to fail on mismatch"
```

Comment:
Unless this `"ERR"` is here the string comes out as `"The: responses from the Tee subchain and down-chain did not match and behavior is set to fail on mismatch"`.

Setting `"ERR: "` will cause the string to come out as `"ERR:: responses from the Tee subchain and down-chain did not match and behavior is set to fail on mismatch"`.

Reply:
Is this a redis protocol thing?
Maybe you should write a comment in the code explaining whats going on.

Reply:
Hmm... this isnt a redis specific transform, I'm quite confused what you mean then.

Reply:
The first byte of an error response from Redis is a minus sign followed by the type of error: e.g. `"-ERR"` for a generic error. Our Redis codec is taking this, thinking the error type is `"The"` and serialising it as `"-The responses from the Tee subchain and down-chain did not match and behavior is set to fail on mismatch"`. The Redis client receives the error response, thinks it's type is `"The"` as well and formats it like `"{Error type}: message"`

This can be fixed by changing `fn build_redis_response_frame` to add `"ERR "` to the front of our generic errors.

Reply:
nice find!
Should the change maybe go in redis-protocol?

Reply:
I don't think so, the issue is how we are converting our generic `QueryResponse` errors to Redis ones.

Reply:
ah, doesnt look like it.
My only concern then is if anything is using the current functionality of  build_redis_response_frame to set a custom error type.

Reply:
https://github.com/shotover/shotover-proxy/blob/b029e97a1b03ed0115d85c51c650f119013d834d/shotover-proxy/src/protocols/redis_codec.rs#L405 Hmm looks like we are passing redis errors straight through like this.

Reply:
So I think the code as you've currently implemented it is the correct approach.
Thanks for investigating!

Reply:
For now, but when we need to handle Cassandra errors too it won't work.

Reply:
yeah, thats something to look into when we get to the cassandra stuff.